src/main/java/rx/internal/operators/OperatorMerge.java

```diff
@@ -246,63 +246,58 @@ private void handleScalarSynchronousObservable(ScalarSynchronousObservable<? ext
 
         private void handleScalarSynchronousObservableWithoutRequestLimits(ScalarSynchronousObservable<? extends T> t) {
             T value = t.get();
-            if (getEmitLock()) {
-                boolean moreToDrain;
-                try {
-                    actual.onNext(value);
-                } finally {
-                    moreToDrain = releaseEmitLock();
-                }
-                if (moreToDrain) {
-                    drainQueuesIfNeeded();
-                }
-                request(1);
-                return;
-            } else {
-                try {
-                    getOrCreateScalarValueQueue().onNext(value);
-                } catch (MissingBackpressureException e) {
-                    onError(e);
+            try {
+                synchronized (this) {
+                    if (emitLock) {
+                        missedEmitting = true;
+                        getOrCreateScalarValueQueue().onNext(value);
+                        return;
+                    }
+                    missedEmitting = false;
+                    emitLock = true;
                 }
+            } catch (MissingBackpressureException e) {
+                onError(e);
                 return;
             }
+            boolean moreToDrain;
+            try {
+                actual.onNext(value);
+            } finally {
+                moreToDrain = releaseEmitLock();
+            }
+            if (moreToDrain) {
+                drainQueuesIfNeeded();
+            }
+            request(1);
         }
 
         private void handleScalarSynchronousObservableWithRequestLimits(ScalarSynchronousObservable<? extends T> t) {
+            
+            // if we didn't return above we need to enqueue
+            // enqueue the values for later delivery
+            try {
+                getOrCreateScalarValueQueue().onNext(t.get());
```

Comment:
This now means it will always pass through a queue which diminishes some of the optimization. This class tries hard to avoid putting values in a queue if it can otherwise emit directly.